handbook/direction/code-graph/code-insights/index.md

```diff
@@ -1,44 +1,220 @@
+> Sourcegraph Code Insights provide Sourcegraph-built custom value metrics and combines with all other data sources about your code to be the singular home for any high-level question, impact, or code analysis.
+
+#### 10 year vision
+
+> Code Insights is the foundation upon which all developers everywhere naturally track, measure, learn, predict, and optimize their engineering and codebase – the best practices of big data become the best practices for managing big code.
+
+### Guiding principles
+
+### Code Insights is built first for engineering leaders
+
+The answer to "why are building Code Insights _now_, ahead of other cool projects Sourcegraph can pursue?" is that Code Insights serves the needs of engineering leaders.
+
+Engineer leaders may not need to use Sourcegraph directly for code search (though their teams do), but they do need to answer questions about their code that Sourcegraph, via Code Insights, is well-positioned to serve.
+
+**What this looks like:**
+
+- If individual developers want feature A, and engineering leaders want feature B, we'll likely build feature B first (but that does not mean we won't later build feature A, especially as the product matures)
+- As we collect feedback and case studies towards our GA launch, we are _particularly_ focused on engineering leaders
+
+**What this does _not_ look like:**
+
+- If engineering leaders want feature C, but feature C would be _harmful_ to the experience of individual developers, we will not build feature C.
+
+### Code Insights layers together the Sourcegraph feature ecosystem
+
+Code Insights naturally sits at the intersection of many other key features, and there are validated needs for, for example: customers to have Insights visualizing what they monitor with Code Monitors, Batch Changes to naturally flow into Code Insights for tracking progress and impact, Precise Code Intelligence that powers code insights for any specific symbol across all repositories.
+
+As Sourcegraph becomes an increasingly powerful developer tool, Code Insights will be vital to tracking and understanding how both Sourcegraph and a company's engineers impact a codebase.
+
+**What this looks like:**
+
+- Code Insights will be extensible and integrate with other Sourcegraph features
+
+**What this does _not_ look like:**
+
+- We won't add integrations just for the sake of integrations, without a validated need and use case
+
+### Code Insights is NOT for "tracking individual developers"
+
+Code Insights is not built for trying to track an individual's output and whether or not someone is a "good" engineer or "high performing" employee. There are (unfortunately) other tools that exist that have made the mistake of tracking easy-to-find, pointless-to-measure metrics like "number of commits" or "number of lines of code" an individual contributes.
```

Comment:
Should we soften the language here?

Reply:
We totally could, I'm open to it. Something like this?
```suggestion
Code Insights is not built for trying to track an individual's output and whether or not someone is a "good" engineer or "high performing" employee. There are other tools that exist that track what (we think) are relatively low-signal basic git metrics like like "number of commits" or "number of lines of code" an individual contributes.
```

Reply:
@thisisvic let me know what you're thinking – would love to get this merged in on Wednesday if possible!